pilot/pkg/xds/eds.go

```diff
@@ -377,13 +378,24 @@ type EndpointBuilder struct {
 	destinationRule *model.Config
 	service         *model.Service
 
-	// These fields are provided for convenience only
+	// These fields are provided for convenience and pre-computation only
 	subsetName string
 	hostname   host.Name
 	port       int
 	push       *model.PushContext
 }
 
+func (b EndpointBuilder) Key() string {
+	params := []string{b.clusterName, b.network, b.clusterID, util.LocalityToString(b.locality)}
```

Comment:
nit: need only one of clusterName and clusterID

Reply:
clusterid refers to k8s cluster, name refers to envoy cluster. I'll add a comment